certbot/renewal.py

```diff
@@ -109,7 +109,9 @@ def _restore_webroot_config(config, renewalparams):
     if "webroot_map" in renewalparams:
         if not cli.set_by_cli("webroot_map"):
             config.webroot_map = renewalparams["webroot_map"]
-    elif "webroot_path" in renewalparams:
+    # To understand why webroot_path and webroot_map processing are not mutually exclusive, see:
+    # https://github.com/certbot/certbot/pull/7095
+    if "webroot_path" in renewalparams and not renewalparams.get('webroot_map', {}):
```

Comment:
Can we simplify this old code by removing the `and not renewalparams...` condition (and the `logger.debug` call below) and always restore the `webroot_path` if it exists? If not, why not?

Reply:
Someone took great care here to make this branch be executed on specific conditions, and for legacy support only. I do not know why, but since I have no spec here, I prefer to limit at the maximum the deviation from the previous behavior.

Adding the condition is sufficient to solves the bug, so I would prefer to keep it and reduce the risk of other regressions (maybe from logic outside of Certbot) that could be created by always propagating both parameters.

If we remove the logger, we change from a legacy situation to a standard. Why changing that?

Reply:
I believe the previous behavior was unnecessary, overly complex, and when combined with other changes, created this problem in the first place. We could potentially leave this to be fixed "later", but since we're here I think we should take the time to investigate and do it properly. I already have looked into it and think what I propose here should work, but I'd want a 2nd set of eyes on that decision. Do you feel comfortable doing that?

Regardless of any arguments for code health though, I now think this change this isn't sufficient. Follow my instructions at https://github.com/certbot/certbot/issues/7048 but use two webroot paths. (If you haven't seen this before, you can run `certbot certonly -w /first/webroot/path -d domain.at.first.path -w /second/webroot/path -d domain.at.second.path`.) Combing that with my instructions at https://github.com/certbot/certbot/issues/7048 caused information on the 2nd webroot path to be dropped with this patch.

Reply:
Implemented, as discussed.